api/metadata_handler_medium_test.go

```diff
@@ -0,0 +1,133 @@
+	w := httptest.NewRecorder()
+	client := &http.Client{}
+
+	metadataHandler := MetadataHandler{nil, client,""}
+	metadataHandler.ServeHTTP(w, r)
+
+	assert.Equal(t, http.StatusBadRequest, w.Code)
+}
+
+func TestFilterMetadataSearchHandler_Success(t *testing.T) {
+	handler := func(w http.ResponseWriter, r *http.Request) {
+		http.ServeFile(w, r, "../shared/metadata_testdata/gzip_testfile.tar.gz")
+	}
+	server := httptest.NewServer(http.HandlerFunc(handler))
+	defer server.Close()
+
+	body :=
+	`{
+		"run_ids": [0, 1, 2],
+		"query": {
+			"exists": [{
+				"link": "bugs.chromium.org"
+			}]
+		}
+	}`
+	bodyReader := strings.NewReader(body)
+	r := httptest.NewRequest("POST", "/abd/api/metadata?product=chrome&product=safari", bodyReader)
+	w := httptest.NewRecorder()
+	client := &http.Client{}
+
+	metadataHandler := MetadataSearchHandler{nil, client, server.URL}
+	metadataHandler.ServeHTTP(w, r)
+
+	assert.Equal(t, http.StatusOK, w.Code)
+	res := w.Body.String()
+	assert.Equal(t, "[{\"test\":\"/IndexedDB/bindings-inject-key.html\",\"urls\":[\"bugs.chromium.org/p/chromium/issues/detail?id=934844\",\"\"]},{\"test\":\"/html/browsers/history/the-history-interface/007.html\",\"urls\":[\"bugs.chromium.org/p/chromium/issues/detail?id=592874\",\"\"]}]", res)
+}
+
+func TestFilterMetadataSearchHandler_MissingProducts(t *testing.T) {
+	body :=
+	`{
+		"run_ids": [0, 1, 2],
+		"query": {
+			"exists": [{
+				"link": "bugs.chromium.org"
+			}]
+		}
+	}`
+	bodyReader := strings.NewReader(body)
+	r := httptest.NewRequest("GET", "/abd/api/metadata?", bodyReader)
+	w := httptest.NewRecorder()
+	client := &http.Client{}
+
+	metadataHandler := MetadataSearchHandler{nil, client,""}
+	metadataHandler.ServeHTTP(w, r)
+
+	assert.Equal(t, http.StatusBadRequest, w.Code)
+}
+
+func TestFilterMetadataSearchHandler_NotLink(t *testing.T) {
+	body :=
+	`{
+		"run_ids": [0, 1, 2],
+		"query": {
+			"exists": [{
+				"pattern": "bugs.chromium.org"
+			}]
+		}
+	}`
+	bodyReader := strings.NewReader(string(body))
+	r := httptest.NewRequest("POST", "/abd/api/metadata?product=chrome&product=safari", bodyReader)
+	w := httptest.NewRecorder()
+	client := &http.Client{}
+
+	metadataHandler := MetadataSearchHandler{shared.NewNilLogger(), client, ""}
+	metadataHandler.ServeHTTP(w, r)
+
+	assert.Equal(t, http.StatusInternalServerError, w.Code)
+}
+
+func TestFilterMetadata(t *testing.T) {
+	metadata := shared.MetadataResults(shared.MetadataResults{shared.MetadataResult{Test: "/foo/bar/b.html", URLs: []string{"", "https://aa.com/item", "https://bug.com/item"}}, shared.MetadataResult{Test: "bar", URLs: []string{"", "https://external.com/item", ""}}})
+	abstractLink := query.AbstractLink{Pattern: "bug.com"}
+
+	res := filterMetadata(abstractLink, metadata)
+
+	assert.Equal(t, 1, len(res))
+	assert.Equal(t, "/foo/bar/b.html", res[0].Test)
+	assert.Equal(t, "", res[0].URLs[0])
+	assert.Equal(t, "https://aa.com/item", res[0].URLs[1])
+	assert.Equal(t, "https://bug.com/item", res[0].URLs[2])
+}
```

Comment:
Nit: This file is missing a newline at EOF. Again, try to make sure `gofmt` is working in your VSCode.